src/config_format/flb_cf_yaml.c

```diff
@@ -1338,114 +1408,252 @@ static char *get_real_path(char *file, char *path, size_t size)
+    return 0;
+}
+
+static int state_create_group(struct flb_cf *cf, struct parser_state *state, char *name)
+{
+    state->cf_group = flb_cf_group_create(cf, state->cf_section,
+                                          "processors", strlen("processors"));
+    if (state->cf_group == NULL) {
+        flb_error("unable to create new processors section group");
+        return YAML_FAILURE;
+    }
+    return YAML_SUCCESS;
+}
+
+static struct parser_state *state_pop(struct local_ctx *ctx)
+{
+    struct parser_state *last;
+
+    if (cfl_list_is_empty(&ctx->states)) {
         return NULL;
     }
+    last = cfl_list_entry_last(&ctx->states, struct parser_state, _head);
+    cfl_list_del(&last->_head);
 
-    /* resolve real path for caller file and target file */
-#ifndef FLB_HAVE_STATIC_CONF
-    if (caller_file) {
-        p = get_real_path(caller_file, caller_path, PATH_MAX + 1);
-        if (!p) {
-            state_destroy(s);
-            return NULL;
-        }
-        s->caller_file = flb_sds_create(caller_file);
-        s->caller_root_path = flb_sds_create(caller_path);
+    if (last->allocation_flags & HAS_KEY) {
+        flb_sds_destroy(last->key);
     }
-    else {
-        s->caller_file = flb_sds_create(s->file);
-        s->caller_root_path = flb_sds_create(s->root_path);
+    if (last->allocation_flags & HAS_KEYVALS) {
+        cfl_kvlist_destroy(last->keyvals);
     }
+    state_destroy(last);
 
-    /* check if the file exists */
-    ret = stat(file, &st);
-    if (ret == 0) {
-        p = get_real_path(file, file_path, PATH_MAX + 1);
-        s->file = flb_sds_create(file);
-        s->root_path = flb_sds_create(file_path);
+    if (cfl_list_is_empty(&ctx->states)) {
+        return NULL;
     }
-    else if (errno == ENOENT && caller_file && s->caller_root_path != NULL) {
-        snprintf(file_path, PATH_MAX, "%s/%s", s->caller_root_path, file);
-        s->file = flb_sds_create(file_path);
+    return cfl_list_entry_last(&ctx->states, struct parser_state, _head);
+}
+
+static void state_destroy(struct parser_state *s)
+{
+    flb_free(s);
+}
+
+static struct parser_state *state_create(struct file_state *parent, struct file_state *file)
+{
+    struct parser_state *state;
+
+    /* allocate context */
+    state = flb_calloc(1, sizeof(struct parser_state));
+    if (!state) {
+        flb_errno();
+        return NULL;
+    }
+
+    state->file = file;
+#ifndef FLB_HAVE_STATIC_CONF
+    if (parent) {
+       state->file->parent = parent;
     }
+#else
+    s->file->name = flb_sds_create("***static***");
+    s->file->path = flb_sds_create("***static***");
 #endif
 
-    return s;
+    return state;
 }
 
 static int read_config(struct flb_cf *cf, struct local_ctx *ctx,
-                       char *caller_file, char *cfg_file)
+                       struct file_state *parent, char *cfg_file)
 {
     int ret;
     int status;
     int code = 0;
-    char *file;
     struct parser_state *state;
+    flb_sds_t file = NULL;
```

Comment:
Fix these names

Reply:
They look fine to me.

Reply:
Does `file` sound like an intuitive or self explanatory name that would make it easier for someone else?
These names should clearly convey what they represent so if at some point either you or someone else needs to make sense of this code to expand or debug it they don't waste time or introduce a bug.